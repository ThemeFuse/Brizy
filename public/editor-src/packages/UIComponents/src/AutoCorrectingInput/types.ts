import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
