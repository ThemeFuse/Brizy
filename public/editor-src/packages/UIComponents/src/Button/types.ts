import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
