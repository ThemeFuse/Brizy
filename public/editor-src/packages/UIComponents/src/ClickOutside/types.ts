import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
