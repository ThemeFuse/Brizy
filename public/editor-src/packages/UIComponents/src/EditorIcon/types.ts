export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
