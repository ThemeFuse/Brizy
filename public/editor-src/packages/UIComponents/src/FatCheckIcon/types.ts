import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
