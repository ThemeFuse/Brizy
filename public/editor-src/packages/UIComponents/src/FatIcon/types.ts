import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
