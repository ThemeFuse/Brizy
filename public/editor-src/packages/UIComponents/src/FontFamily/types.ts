import { FontFamilyType } from "../types/fonts/familyType";
import { Weight } from "../types/fonts/weight";

export type FontsBlock = Partial<{
  config: Array<FontObject>;
  blocks: Array<FontObject>;
  [FontFamilyType.google]: Array<FontObject>;
  [FontFamilyType.upload]: Array<FontObject>;
}>;

export interface Props {
  addFont?: VoidFunction;
  onChange: (font: FontWithType) => void;
  value: string;
  className?: string;
  addFontLabel?: string;
  fonts: FontsBlock;
}
export interface FontSizes {
  default: string;
  great_vibes: string;
  alex_brush: string;
  allura: string;
  parisienne: string;
}

export interface FontObject {
  brizyId?: string;
  family: string;
  id: string;
  title: string;
  weights: Array<Weight>;
}

export interface FontWithType {
  id: string;
  type: FontFamilyType;
  weights: Array<Weight>;
}

export interface FontList {
  fonts: Array<FontObject>;
  type: FontFamilyType;
}
