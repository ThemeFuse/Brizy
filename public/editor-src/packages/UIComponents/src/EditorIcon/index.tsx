import Loadable from "@loadable/component";
import classnames from "classnames";
import React, { CSSProperties, FC, MouseEvent, ReactElement } from "react";
import { IconsName } from "./types";

const LoadableIcon = Loadable(async () => {
  const { Icon } = await import("./components/Icon");
  return Icon;
});
export interface Props {
  className?: string;
  icon: IconsName;
  style?: CSSProperties;
  onClick?: (e: MouseEvent<SVGElement>) => void;
}

export const EditorIcon: FC<Props> = ({
  className,
  icon,
  style,
  onClick
}): ReactElement => {
  const _className = classnames(
    "brz-icon-svg brz-ed-icon-svg align-[initial]",
    className
  );

  return (
    <LoadableIcon
      name={icon}
      className={_className}
      style={style}
      onClick={onClick}
    />
  );
};
