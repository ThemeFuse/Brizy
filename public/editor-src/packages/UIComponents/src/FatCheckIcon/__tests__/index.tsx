import { fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";
import { FatCheckIcon } from "../";
import { IconsName } from "../../EditorIcon/types";

describe("Test props", () => {
  test("Put label text on container element as title attribute", async () => {
    const { container } = render(
      <FatCheckIcon icon={IconsName.Settings} label="Test label" />
    );

    expect(container.firstChild).toMatchSnapshot();

    expect(container.firstElementChild?.getAttribute("title")).toBe(
      "Test label"
    );

    expect(
      container.querySelector(".brz-ed--fat-icon__wrapper--inactive")
    ).toBeInTheDocument();

    expect(
      container.querySelector(".brz-ed--fat-icon__label--inactive")
    ).toBeInTheDocument();

    expect(
      container.querySelector(".brz-ed--check-mark__unchecked")
    ).toBeInTheDocument();

    await waitFor(() =>
      expect(
        container.querySelector(".brz-icon-svg__inactive")
      ).toBeInTheDocument()
    );
  });

  test("should have onClick", () => {
    const onClick = jest.fn();
    const { container } = render(
      <FatCheckIcon
        icon={IconsName.Settings}
        label="Test label"
        onClick={onClick}
      />
    );
    const div = container.querySelector(".brz-ed--fat-icon");

    div && fireEvent.click(div);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("should have onCheck", () => {
    const onCheck = jest.fn();
    const { container } = render(
      <FatCheckIcon
        icon={IconsName.Settings}
        label="Test label"
        onCheck={onCheck}
      />
    );

    const checkMark = container.querySelector(".brz-ed--check-mark");

    checkMark && fireEvent.click(checkMark);

    expect(onCheck).toHaveBeenCalledTimes(1);
  });

  test("should be isChecked true", () => {
    const { container } = render(
      <FatCheckIcon icon={IconsName.Settings} label="Test label" isChecked />
    );

    expect(container.firstChild).toMatchSnapshot();

    expect(
      container.querySelector(".brz-ed--check-mark__checked")
    ).toBeInTheDocument();
  });

  test("should be isActive true", async () => {
    const { container } = render(
      <FatCheckIcon icon={IconsName.Settings} label="Test label" isActive />
    );

    expect(container.firstChild).toMatchSnapshot();

    expect(
      container.querySelector(".brz-ed--fat-icon__wrapper--active")
    ).toBeInTheDocument();

    expect(
      container.querySelector(".brz-ed--fat-icon__label--active")
    ).toBeInTheDocument();

    await waitFor(() =>
      expect(
        container.querySelector(".brz-icon-svg__active")
      ).toBeInTheDocument()
    );
  });

  test("className should be className", () => {
    const { container } = render(
      <FatCheckIcon
        icon={IconsName.Settings}
        label="Test label"
        className="className"
      />
    );

    expect(container.firstChild).toMatchSnapshot();

    expect(container.querySelector(".brz-ed--fat-icon")).toHaveClass(
      "className"
    );
  });
});
