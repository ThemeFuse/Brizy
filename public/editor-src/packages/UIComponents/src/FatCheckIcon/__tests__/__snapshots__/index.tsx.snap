// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test props Put label text on container element as title attribute 1`] = `
<div
  class="brz-ed--fat-icon"
  title="Test label"
>
  <div
    class="brz-ed--fat-icon__wrapper brz-ed--fat-icon__wrapper--inactive"
  >
    <span
      class="brz-ed--check-mark brz-ed--fat-check-icon__check brz-ed--check-mark__unchecked"
    />
  </div>
  <div
    class="brz-ed--fat-icon__label brz-ed--fat-icon__label--inactive"
  >
    Test label
  </div>
</div>
`;

exports[`Test props className should be className 1`] = `
<div
  class="className brz-ed--fat-icon"
  title="Test label"
>
  <div
    class="brz-ed--fat-icon__wrapper brz-ed--fat-icon__wrapper--inactive"
  >
    <span
      class="brz-ed--check-mark brz-ed--fat-check-icon__check brz-ed--check-mark__unchecked"
    />
  </div>
  <div
    class="brz-ed--fat-icon__label brz-ed--fat-icon__label--inactive"
  >
    Test label
  </div>
</div>
`;

exports[`Test props should be isActive true 1`] = `
<div
  class="brz-ed--fat-icon"
  title="Test label"
>
  <div
    class="brz-ed--fat-icon__wrapper brz-ed--fat-icon__wrapper--active"
  >
    <span
      class="brz-ed--check-mark brz-ed--fat-check-icon__check brz-ed--check-mark__unchecked"
    />
  </div>
  <div
    class="brz-ed--fat-icon__label brz-ed--fat-icon__label--active"
  >
    Test label
  </div>
</div>
`;

exports[`Test props should be isChecked true 1`] = `
<div
  class="brz-ed--fat-icon"
  title="Test label"
>
  <div
    class="brz-ed--fat-icon__wrapper brz-ed--fat-icon__wrapper--inactive"
  >
    <span
      class="brz-ed--check-mark brz-ed--fat-check-icon__check brz-ed--check-mark__checked"
    />
  </div>
  <div
    class="brz-ed--fat-icon__label brz-ed--fat-icon__label--inactive"
  >
    Test label
  </div>
</div>
`;
